bot/cogs/antimalware.py

```diff
@@ -0,0 +1,43 @@
+import logging
+
+from discord import Message, utils
+from discord.ext.commands import Bot, Cog
+
+from bot.constants import AntiMalware as AntiMalwareConfig, Channels
+
+log = logging.getLogger(__name__)
+
+
+class AntiMalware(Cog):
+    """Cog providing anti-malware behavior."""
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        self.whitelist = tuple(AntiMalwareConfig.whitelist)
+
+    @Cog.listener()
+    async def on_message(self, message: Message) -> None:
+        """Identify messages with prohibited attachments."""
+        log.trace("Entered AntiMalware.on_message()")
+        rejected_attachments = [a for a in message.attachments if
+                                not a.filename.lower().endswith(self.whitelist)]
+        detected_pyfile = len([a for a in message.attachments if a.filename.lower().endswith('.py')]) > 0
+
+        if len(rejected_attachments) > 0:
+            log.trace("Identified rejected attachment(s)")
+            # Send a message indicating the problem to the user (with special treatment for .py)
+            author = message.author
+            if detected_pyfile:
+                msg = f"{author.mention}, it looks like you tried to attach a Python file - please " \
+                    f"use a code-pasting service such as https://paste.pythondiscord.com/ instead."
+            else:
```

Comment:
if there can be several attachements, why use a `if else` structure?

Reply:
I didn't want to clutter things up with too much response text, and deemed the pastebin advice to be the more important of the two. Just a personal judgment call, I'll be happy to revise if requested by staff.

Reply:
I agree with Ben.